python/ray/train/_internal/dl_predictor.py

```diff
@@ -83,3 +83,21 @@ def _predict_pandas(
                 {"predictions": TensorArray(self._tensor_to_array(output))},
                 columns=["predictions"],
```

Comment:
I think this was an oversight before, but we should just be able to call `convert_batch_type_to_pandas` here.

`_predict_pandas` implementation can just be

```
numpy_input = convert_pandas_to_batch_type(BatchType.NUMPY)
numpy_output = self._predict_numpy(numpy)
return convert_batch_type_to_pandas(numpy_output)
```

This should keep the implementation very simple

Reply:
this makes sense at high level, for prediction we will need a column to hold prediction outputs and join with inputs tho, which `convert_batch_type_to_pandas` won't provide out of the box. I can call this func and add the prediction column if it doesn't exist.

Reply:
`numpy_output` in the example above should already contain a column name, no? https://github.com/ray-project/ray/pull/28917/files/9234923a3789910d68f6464a1c363a4c9a480f7e#diff-262a0de6b46cb9213bc50a6ff3d4a05232d6d24ae049ab9be2a152cf85c1e573R103